scripts/service_install.sh

```diff
@@ -4,6 +4,7 @@ set -euxo pipefail
 service_name="gatestext.service"
 
 dir=$(pwd)
+cp $dir/bin/gatestext_script_linux /usr/local/bin
 eval "sed s#{WORKDIR}#$dir# $dir/services/gatestext.service > /lib/systemd/system/gatestext.service"
-chmod 755 /lib/systemd/system/$service_name
-systemctl daemon-reload
\ No newline at end of file
+chmod 644 /lib/systemd/system/$service_name
```

Comment:
As per [this doc on redhat systemd files](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/chap-managing_services_with_systemd), the location might need to be changed - 

<html>
<body>
<!--StartFragment--><p class="title" style="box-sizing: border-box; margin: 0px 0px 1.8em; line-height: 1.5em; color: rgb(37, 37, 37); font-family: &quot;Red Hat Text&quot;, RedHatText, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong style="box-sizing: border-box; font-weight: 700;">Table 10.2. Systemd Unit Files Locations</strong></p><div class="table-contents" style="box-sizing: border-box; overflow-x: auto; width: 50vw; max-width: 75rem; color: rgb(37, 37, 37); font-family: &quot;Red Hat Text&quot;, RedHatText, &quot;Helvetica Neue&quot;, Arial, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Directory | Description
-- | --
/usr/lib/systemd/system/ | Systemd unit files distributed with installed RPM packages.
/run/systemd/system/ | Systemd unit files created at run time. This directory takes precedence over the directory with installed service unit files.
/etc/systemd/system/ | Systemd unit files created by systemctl enable as well as unit files added for extending a service. This directory takes precedence over the directory with runtime unit files.

</div><!--EndFragment-->
</body>
</html>

Reply:
For ubuntu, the path you provided seems correct, but for redhat its different. 

From [this post](https://askubuntu.com/questions/876733/where-are-the-systemd-units-services-located-in-ubuntu) - 

> The package-provided service files are all usually located in /lib/systemd/system. For example, search for .service in the package index.
From man systemd.unit:
/etc/systemd/system/*
/run/systemd/system/*
/lib/systemd/system/*
...

Given the discrepancies between multiple distros, please devise a way to make this path distro dependent vs just a static path. Thanks!

Reply:
I verified for ubuntu and did work. At this point enabled distros still works with /lib/systemd/system static path since redhat distros have symlink `/lib is a symlink to /usr/lib`.